indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -1887,10 +1887,12 @@ public Response pauseHTTP(
   @VisibleForTesting
   public Response pause() throws InterruptedException
   {
-    if (!(status == Status.PAUSED || status == Status.READING)) {
+    Status currentStatus = status;
```

Comment:
```suggestion
    // Read the volatile status into a variable so that its value does not change while the condition is evaluated
    final Status currentStatus = status;
```